File: Data Structures & Algorithms/decode-ways/submission-0.cpp
class Solution {
public:
    int numDecodings(string s) {
        if(s.size() == 0) return 1;
        if(s[0]=='0') return 0;
        if(s.size() == 1) return 1;
        string s1,s2;
        s1.append(s,1);
        s2.append(s,2);
        if(s.size()>=2){
            if(((s[0]-'0')*10 + (s[1]-'0')) <27)
            return numDecodings(s1) + numDecodings(s2);
        }
        return numDecodings(s1);
    }
};

File: Data Structures & Algorithms/clone-graph/submission-0.py

# Definition for a Node.
class Node:
    def __init__(self, val = 0, neighbors = None):
        self.val = val
        self.neighbors = neighbors if neighbors is not None else []


class Solution:
    def cloneGraph(self, node: Optional['Node']) -> Optional['Node']:
        if not node:
            return None

        clone = {}

        def cl(curr):
            if curr in clone:
                return clone[curr]
            
            copy = Node(curr.val)

            clone[curr] = copy

            for neigh in curr.neighbors:
                copy.neighbors.append(cl(neigh))

            return copy
    
        return cl(node)



